### PR TITLE
Adds possibility to define a goal description

### LINKED ISSUE #7346: Description/Comment on Goals
I experienced a certain problem about goals while working in my company. As some users from the management also have access to Piwik they try to analyze the traffic on our website. We have a lot of goals defined to measure the usage of our website. This goals only have relatively short names and use a certain naming convention so that we as the developers know in witch part of our web application this goals are placed. But this short names often aren't as clear to everybody as we thought and this leads to recurring misinterpretation of the goals. So it would be very helpful if there was the possibility to create a detailed description for goals. I guess this shouldn't be a lot of work, but it would be a huge improvement in usability and collaborative usage of Piwik.

### PR DESCRIPTION
**Pull request by sgiehl**
image

image

image

image

fixes #7346

### CODE REVIEW

File: plugins/Goals/API.php [hunk 1]
```diff
@@ -151,20 +154,23 @@ private function getModel()
      * @param bool $caseSensitive
      * @param bool|float $revenue
      * @param bool $allowMultipleConversionsPerVisit
+     * @param string $description
      * @return void
      */
-    public function updateGoal($idSite, $idGoal, $name, $matchAttribute, $pattern, $patternType, $caseSensitive = false, $revenue = false, $allowMultipleConversionsPerVisit = false)
+    public function updateGoal($idSite, $idGoal, $name, $matchAttribute, $pattern, $patternType, $caseSensitive = false, $revenue = false, $allowMultipleConversionsPerVisit = false, $description = '')
     {
         Piwik::checkUserHasAdminAccess($idSite);
```
Comment (tsteur) [inline plugins/Goals/API.php]: I wonder if we should keep the currently saved `$description` if `$description = false` and clear it if `$description = ''`. `false` could be default but I'm not sure what is better and whether is needed / more intuitive. I guess as `caseSensitive` and other optional parameters works the same as implemented it's all good and consistent the way it is implemented now :+1:

File: plugins/Goals/API.php [hunk 2]
```diff
@@ -101,20 +101,23 @@ public function getGoals($idSite)
      * @param bool|float $revenue If set, default revenue to assign to conversions
      * @param bool $allowMultipleConversionsPerVisit By default, multiple conversions in the same visit will only record the first conversion.
      *                         If set to true, multiple conversions will all be recorded within a visit (useful for Ecommerce goals)
+     * @param string $description
      * @return int ID of the new goal
      */
-    public function addGoal($idSite, $name, $matchAttribute, $pattern, $patternType, $caseSensitive = false, $revenue = false, $allowMultipleConversionsPerVisit = false)
+    public function addGoal($idSite, $name, $matchAttribute, $pattern, $patternType, $caseSensitive = false, $revenue = false, $allowMultipleConversionsPerVisit = false, $description = '')
```
Comment (tsteur) [inline plugins/Goals/API.php]: I'm not sure if there are tests for goals, would be possible to add a simple test maybe?

Comment (tsteur) [inline plugins/Goals/API.php]: Ignore it, saw the test :)

**Comment by tsteur** [comment]
A ui test fails here: http://builds-artifacts.piwik.org/piwik/piwik/goal_description/17418/UIIntegrationTest_goals_manage can you maybe have a look?

**Comment by sgiehl** [comment]
Guess the OmniFixture needs an update, so that the database contains the column. But I would wait until #9272 is merged.

**Comment by tsteur** [comment]
OK :)

**Comment by tsteur** [comment]
FYI: We're waiting for #9272

File: plugins/Goals/templates/getOverviewView.twig [hunk 3]
```diff
@@ -9,7 +9,7 @@
     {% set conversion_rate=goal.conversion_rate %}
     {% set name=goal.name %}
     <div class="goalEntry" style="clear:both">
-        <h2>
+        <h2 {% if goal.description %}title="{{ goal.description }}"{% endif %}>
```
Comment (mattab) [inline plugins/Goals/templates/getOverviewView.twig]: escaping is missing?

**Comment by mattab** [comment]
Targeting for 3.0 instead as it may be a little risky for 2.16.0?

**Comment by sgiehl** [comment]
Replaced by #10057 which targets 3.x branch

**Pull request closed without merging**